core/trino-main/src/test/java/io/trino/server/ui/TestWebUi.java

```diff
@@ -772,6 +811,24 @@ private void assertAuth2Authentication(HttpServerInfo httpServerInfo, String acc
         assertResponseCode(client, getLocation(baseUri, "/ui/unknown"), SC_NOT_FOUND);
         assertResponseCode(client, getLocation(baseUri, "/ui/api/unknown"), SC_NOT_FOUND);
 
+        if (refreshTokensEnabled) {
+            // wait till auth token expires
+            Thread.sleep(Duration.between(ZonedDateTime.now(), authTokenExpiration).toMillis());
```

Comment:
Could you issue an already expired token? To get rid of this the sleep.

Reply:
Yep, done